databricks-kube/src/crds/databricks_secret.rs

```diff
@@ -0,0 +1,174 @@
+use std::collections::BTreeMap;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::time::Duration;
+
+use crate::context::Context;
+use databricks_rust_secrets::apis::secret_api;
+use databricks_rust_secrets::models::{
+    WorkspaceDeleteSecret, WorkspaceGetSecretResponse, WorkspacePutSecret,
+};
+use futures::{Stream, StreamExt, TryFutureExt, TryStreamExt};
+use k8s_openapi::api::core::v1::Secret;
+use k8s_openapi::ByteString;
+use kube::runtime::controller::Action;
+use kube::runtime::finalizer::Event;
+use kube::runtime::reflector::ObjectRef;
+use kube::runtime::{finalizer, watcher, Controller};
+use kube::{core::object::HasSpec, CustomResource, CustomResourceExt};
+use kube::{Api, ResourceExt};
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::error::DatabricksKubeError;
+use crate::traits::rest_config::RestConfig;
+
+#[derive(Clone, CustomResource, Debug, Default, Deserialize, PartialEq, Serialize, JsonSchema)]
+#[kube(
+    group = "com.dstancu.databricks",
+    version = "v1",
+    kind = "DatabricksSecret",
+    derive = "Default",
+    namespaced
+)]
+pub struct DatabricksSecretSpec {
+    pub scope: String,
+    pub secret_name: String,
+}
+
+impl DatabricksSecret {
```

Comment:
This is the controller that runs inside of the top level tokio worker. Every `default_requeue_interval` seconds it writes the secret values into Databricks. Or on delete/create it reconcilles the actions.